kubernetes_env/benchmark/benchmark.py

```diff
@@ -48,62 +117,37 @@ def run_fortio(platform, threads, qps, run_time):
-        fortio_json = json.load(f)
-        data = fortio_json["DurationHistogram"]
-        return data
-
-
-def benchmark_filter(platform, threads, qps, time, file_name, graph_name):
-    fortio_res = run_fortio(platform, threads, qps, time)
-    if fortio_res == util.EXIT_FAILURE:
-        return fortio_res
-    data = get_histogram(file_name)
-    if data is not None:
-        latency_data = data["Data"]
-        response_times = [resp["End"] - resp["Start"] for resp in latency_data]
-        request_counts = [resp["Count"] for resp in latency_data]
-        plot(response_times, request_counts, "Response time (ms)", "Requests",
-             f"Platform: {platform} Threads: {threads} QPS: {qps}", graph_name)
-        return util.EXIT_SUCCESS
-    else:
-        log.error("Error in processing data")
-        return util.EXIT_FAILURE
-
-
-def start_benchmark(filter_dirs, platform, threads, qps, time, file_name,
-                    graph_name):
+def start_benchmark(fortio, filter_dirs, platform, threads, qps, time):
     if kube_env.check_kubernetes_status() != util.EXIT_SUCCESS:
         log.error("Kubernetes is not set up."
                   " Did you run the deployment script?")
         return util.EXIT_FAILURE
 
-    for (idx, filter_dir) in enumerate(filter_dirs):
-        build_res = kube_env.build_filter(filter_dir)
+    for (idx, fd) in enumerate(filter_dirs):
+        build_res = kube_env.build_filter(fd)
 
         if build_res != util.EXIT_SUCCESS:
-            log.error(f"Setting filter failed for {filter_dir}."
+            log.error(f"Building filter failed for {filter_dir}."
                       " Make sure you give the right path")
             return util.EXIT_FAILURE
 
-        filter_res = kube_env.deploy_filter(filter_dir)
+        filter_res = kube_env.refresh_filter(fd)
         if filter_res != util.EXIT_SUCCESS:
-            log.error(f"Setting filter failed for {filter_dir}."
+            log.error(f"Deploying filter failed for {filter_dir}."
                       " Make sure you give the right path")
             return util.EXIT_FAILURE
-
-        benchmark_res = benchmark_filter(platform, threads, qps, time,
-                                         file_name, f"{graph_name}{idx}")
-        if benchmark_res != util.EXIT_SUCCESS:
+        log.info("Benchmarking...")
+        fname = fd.split("/")[-1]
+        fortio_res = run_fortio(platform, threads, qps, time, fname)
+        if fortio_res != util.EXIT_SUCCESS:
             log.error(f"Error benchmarking for {filter_dir}")
             return util.EXIT_FAILURE
 
-        undeploy_res = kube_env.undeploy_filter()
-        if undeploy_res != util.EXIT_SUCCESS:
-            log.error(f"Setting filter failed for {filter_dir}."
-                      " Make sure you give the right path")
-            return util.EXIT_FAILURE
+    dataf = [
+        f for f in os.listdir(DATA_DIR)
+        if os.path.isfile(DATA_DIR.joinpath(f))
```

Comment:
Might want to use [pathlib](https://docs.python.org/3/library/pathlib.html) directly for this instead of os.xyz